app/integration_action.go

```diff
@@ -141,15 +144,18 @@ func (a *App) DoPostActionWithCookie(postId, actionId, userId, selectedOption st
 
 	if response.Update != nil {
 		response.Update.Id = postId
-		response.Update.AddProp("from_webhook", "true")
-		for key, value := range retain {
-			response.Update.AddProp(key, value)
-		}
-		for _, key := range remove {
-			delete(response.Update.Props, key)
-		}
-		if _, appErr = a.UpdatePost(response.Update, false); appErr != nil {
-			return "", appErr
+		if response.Update.Props == nil {
+			response.Update.Props = originalProps
+		} else {
+			for key, value := range retain {
+				response.Update.AddProp(key, value)
+			}
+			for _, key := range remove {
+				delete(response.Update.Props, key)
+			}
+			if _, appErr = a.UpdatePost(response.Update, false); appErr != nil {
```

Comment:
:+1: approach with `Props'.

But there are other "metadata" that needs to be kept: `IsPinned` and `HasReactions`. I'm fine with dealing with this in a different PR, but I want to make sure it's not forgotten.

Reply:
Right. I'd like to take care of it here - this is for 5.11 so we should have time to discuss - the entire list of "protected" values, props or not. I also still have a TODO left for any other prop values we want to protect